### DISCUSSION TITLE
How to initialize a fungible store for a Fungible asset via TypeScript?

### DISCUSSION THREAD

**Discussion by gregnazario** [aptos-labs/aptos-developer-discussions#64] [Questions] [answered]
From unbound on Discord:

Can I have something in typescript that allows the signer on the frontend to quickly initialize a fungiblestore for them of a certain metadata?

**Comment by gregnazario** [comment, marked as answer]
You can call [`0x1::primary_fungible_store::create_primary_store`](https://github.com/aptos-labs/aptos-core/blob/main/aptos-move/framework/aptos-framework/sources/primary_fungible_store.move#L69-L84) if you're looking for a fungible asset, but it's not an entry function on it's own.

Additionally, [`0x1::primary_fungible_store::transfer`]((https://github.com/aptos-labs/aptos-core/blob/main/aptos-move/framework/aptos-framework/sources/primary_fungible_store.move#L141-L153)) will create the primary fungible store, similar to coin.

```ts
    const metadataAddress = "0x12345";
    const recipientAddress = "0x1337";
    const amount = 0;

    const txn = await aptos.transaction.build.simple({
        sender: account.accountAddress,
        data: {
            function: "0x1::primary_fungible_store::transfer",
            typeArguments: ["0x1::object::ObjectCore"],
            functionArguments: [metadataAddress, recipientAddress, amount],
        }
    });
```

Does that answer your question, or are you looking for something else?

**Reply by unbound26z** [reply]
Initially I had a parameter on my modules function that was an Object<FungibleStore> where some funds will get withdrawn from the module. If a user did not have that fungible asset in the account already, they couldn't call that function. I've changed it in the meantime to take an Object<Metadata> and then I create a fungible store for them on chain and deposit in that store.
That code above doesn't exactly fix the issue, but thanks for the help!

**Reply by peacein2022** [reply]
> Initially I had a parameter on my modules function that was an Object where some funds will get withdrawn from the module. If a user did not have that fungible asset in the account already, they couldn't call that function. I've changed it in the meantime to take an Object and then I create a fungible store for them on chain and deposit in that store.
> That code above doesn't exactly fix the issue, but thanks for the help!

Can you share this code to fix the issue?

**Reply by kshitijc1506** [reply]
[`pfs::ensure_primary_store_exists`](https://github.com/aptos-labs/aptos-core/blob/26c0a51b7ace21dbd43220644ac397b74f460130/aptos-move/framework/aptos-framework/sources/primary_fungible_store.move#L59) returns the account's primary store, and creates one and returns it if it doesn't exist.
